plugins/inputs/snmp/snmp.go

```diff
@@ -120,16 +116,21 @@ type Snmp struct {
 	translator Translator
 }
 
+func (s *Snmp) SetTranslator(name string) {
+	s.Translator = name
+}
```

Comment:
What about calling this `InitTranslator` and initiating the translator, returning error on failure?

Reply:
This is kind of an ugly hack to share the agent's translator setting with the plugins. I didn't want to add more error handling code to the agent to handle an error where it calls this function. There's already a well defined way for inputs to report config errors to the agent- Init func's return value.